test/integration/integration_test.go

```diff
@@ -0,0 +1,208 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+package integration
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"testing"
+
+	"github.com/AzureAD/microsoft-authentication-library-for-go/msal"
+)
+
+// HTTP Helper
+func httpRequest(url string, query map[string]string, accessToken string) ([]byte, error) {
+	request, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, err
+	}
+	request.Header.Set("Authorization", "Bearer "+accessToken)
+	q := request.URL.Query()
+	for key, value := range query {
+		q.Add(key, value)
+	}
+	request.URL.RawQuery = q.Encode()
+	response, err := http.DefaultClient.Do(request)
+	if err != nil {
+		return nil, err
+	}
+	defer response.Body.Close()
+	body, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return nil, err
+	}
+	return body, nil
+}
+
+// Lab Interface
+
+type labClient struct {
+	labApplication *msal.ConfidentialClientApplication
+}
+
+// TODO : Add app object
+
+type user struct {
+	AppID            string `json:"appId"`
+	ObjectID         string `json:"objectId"`
+	UserType         string `json:"userType"`
+	DisplayName      string `json:"displayName"`
+	Licenses         string `json:"licences"`
+	Upn              string `json:"upn"`
+	Mfa              string `json:"mfa"`
+	ProtectionPolicy string `json:"protectionPolicy"`
+	HomeDomain       string `json:"homeDomain"`
+	HomeUPN          string `json:"homeUPN"`
+	B2cProvider      string `json:"b2cProvider"`
+	LabName          string `json:"labName"`
+	LastUpdatedBy    string `json:"lastUpdatedBy"`
+	LastUpdatedDate  string `json:"lastUpdatedDate"`
+	Password         string
+}
+type secret struct {
+	Secret string `json:"value"`
```

Comment:
this "stutters" on use e.g. secret.Secret. Maybe call the field value? so use looks like `secret.Value`. a struct that represents a secret that holds the value